### PR TITLE
LWDEV-9284 Add swagger method to turn on/off maintenance mode

### PR DESCRIPTION
**Pull request by rocketmannot**

### CODE REVIEW

File: src/main/kotlin/com/lykke/matching/engine/services/ApplicationSettingsServiceImpl.kt [hunk 1]
```diff
@@ -22,62 +29,87 @@ class ApplicationSettingsServiceImpl(private val settingsDatabaseAccessor: Setti
...
+                deleteSettingRequestDto.comment,
+                deleteSettingRequestDto.user))
     }
 
     @Synchronized
     override fun deleteSetting(settingsGroup: AvailableSettingGroup, settingName: String, deleteSettingRequestDto: DeleteSettingRequestDto) {
-        val settingToBeDeleted = settingsDatabaseAccessor.getSetting(settingsGroup.settingGroupName, settingName) ?:
+        val deletedSetting = settingsDatabaseAccessor.getSetting(settingsGroup, settingName) ?: return
+
+        val settingToBeDeleted = settingsDatabaseAccessor.getSetting(settingsGroup, settingName) ?:
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/matching/engine/services/ApplicationSettingsServiceImpl.kt]: As i understand this call is redundant. And if it is so then the class SettingNotFoundException can be removed.

Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/services/ApplicationSettingsServiceImpl.kt]: removed `deletedSetting` , logic  with SettingNotFoundException should be kept

File: src/main/kotlin/com/lykke/matching/engine/database/azure/AzureSettingsDatabaseAccessor.kt [hunk 2]
```diff
@@ -117,19 +117,19 @@ class AzureSettingsDatabaseAccessor(connectionString: String, configTableName: S
         if (CollectionUtils.isEmpty(settings)) {
             return null
         }
-        return SettingsGroup(settings.first().partitionKey, toSettings(settings))
+        return SettingsGroup(AvailableSettingGroup.getBySettingsGroupName(settings.first().partitionKey), toSettings(settings))
     }
 
     private fun toSettingsGroup(groupToSettings: Map<String, List<AzureAppSetting>>): Set<SettingsGroup> {
-        return groupToSettings.map { entry -> SettingsGroup(entry.key, toSettings(entry.value)) }.toSet()
+        return groupToSettings.map { entry -> SettingsGroup(AvailableSettingGroup.getBySettingsGroupName(entry.key), toSettings(entry.value)) }.toSet()
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/database/azure/AzureSettingsDatabaseAccessor.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/services/events/listeners/MessageProcessingSwitchListener.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+        val message = getLogMessageMessage(START_ACTION, deleteSettingEvent.user, deleteSettingEvent.comment)
+        LOGGER.info(message)
+        METRICS_LOGGER.logWarning(message)
+    }
+
+    @EventListener
+    private fun messageProcessingSwtichGroupRemoved(deleteSettingGroupEvent: DeleteSettingGroupEvent) {
+        if (deleteSettingGroupEvent.settingGroup != AvailableSettingGroup.MESSAGE_PROCESSING_SWITCH ||
+                deleteSettingGroupEvent.deletedSettings.find { it.enabled } == null) {
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/services/events/listeners/MessageProcessingSwitchListener.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/services/events/listeners/MessageProcessingSwitchListener.kt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            return
+        }
+
+        val message = getLogMessageMessage(START_ACTION, deleteSettingEvent.user, deleteSettingEvent.comment)
+        LOGGER.info(message)
+        METRICS_LOGGER.logWarning(message)
+    }
+
+    @EventListener
+    private fun messageProcessingSwtichGroupRemoved(deleteSettingGroupEvent: DeleteSettingGroupEvent) {
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/services/events/listeners/MessageProcessingSwitchListener.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/config/spring/DatabaseAccessorConfig.kt [hunk 5]
```diff
@@ -172,7 +169,7 @@ open class DatabaseAccessorConfig {
     }
 
     @Bean
-    open fun settingsHistoryDatabaseAccessor(@Value("\${azure.settings.history.database.accessor.table}") tableName: String): SettingsHistoryDatabaseAccessor {
+    open fun aettingsHistoryDatabaseAccessor(@Value("\${azure.settings.history.database.accessor.table}") tableName: String): SettingsHistoryDatabaseAccessor {
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/config/spring/DatabaseAccessorConfig.kt]: fixed

**Pull request merged**